scripts/normalize-by-median.py

```diff
@@ -308,24 +354,16 @@ def main():  # pylint: disable=too-many-branches,too-many-statements
                                          args.n_tables)
 
     input_filename = None
+    global index
+    index = 0
 
-    for index, input_filename in enumerate(args.input_filenames):
+    for f in SavingTable(args.input_filenames, args.dump_frequency, htable,
+                         args.savetable):
         total_acc, discarded_acc, corrupt_files = \
             normalize_by_median_and_check(
-                input_filename, htable, args.single_output_file,
+                f, htable, args.single_output_file,
                 args.fail_save, args.paired, args.cutoff, args.force,
-                corrupt_files, report_fp)
-
-        if (args.dump_frequency > 0 and
-                index > 0 and index % args.dump_frequency == 0):
-            print 'Backup: Saving k-mer counting file through', input_filename
-            if args.savetable:
-                hashname = args.savetable
-                print '...saving to', hashname
-            else:
-                hashname = 'backup.ct'
-                print 'Nothing given for savetable, saving to', hashname
-            htable.save(hashname)
+                report_fp)
 
     if args.paired and args.unpaired_reads:
```

Comment:
Dunno. Not my doing. Looks like it's from the broken paired/`-p` implementation, from https://github.com/dib-lab/khmer/pull/957/files#diff-6b3b058584d44911b3cf3fe21d9c9af4R325

If you meant the mass deletion of the dump frequency code, I moved that into a generator.